File: perl/spada.affy.pl
#!/usr/bin/perl -w
use strict;
use Bio::SeqIO;
use FindBin;
use lib $FindBin::Bin;
use File::Path qw/make_path remove_tree/;
use Data::Dumper;
use List::Util qw/min max sum/;

my $org = "Athaliana";
#$org = "Mtruncatula_3.5";
my $dir = "/home/youngn/zhoup/Data/misc2/spada.affy/$org";

my $f01 = "$dir/01_probe.tbl";
my $f02 = "$dir/02_probe.fa";
#get_probe_seq("$dir/00.tbl", $f01, $f02);

sub get_probe_seq {
    my ($fi, $fo1, $fo2) = @_;
    my ($fhi, $fho);
    open($fhi, "<$fi") or die "cannot read $fi\n";
    open($fho, ">$fo1") or die "cannot write $fo1\n";
    my $seqHO = Bio::SeqIO->new(-file=>">$fo2", -format=>'fasta');
    
    print $fho join("\t", qw/id set seq/)."\n";
    my $h;
    while(<$fhi>) {
        chomp;
        my ($set, $num1, $num2, $num3, $seq) = split "\t";
        $h->{$set} ||= 0;
        $h->{$set} ++;
        my $id = sprintf "%s.%02d", $set, $h->{$set};
        $seqHO->write_seq( Bio::Seq->new(-id=>$id, -seq=>$seq) );
        print $fho join("\t", $id, $set, $seq)."\n";
    }
    $seqHO->close();
    close $fho;
}

File: r/spada.affy.R
library(plyr)
library(data.table)

dir = file.path(DIR_Misc2, "spada.affy", "Athaliana")
te1 = read.table(file.path(dir, "../atdefl_pac.tbl"), sep="\t", header=T, as.is=T)
totalcall = apply(te1[,seq(3,61,by=2)], 1, function(x) "P" %in% x)
te = data.frame(set=te1$Probe.set, call=totalcall)

#fi = file.path(dir, "23.tbl")
fi = file.path(dir, "33.tbl")
#fo = file.path(dir, "25.tbl")
fo = file.path(dir, "35.tbl")
tg = read.table(fi, sep="\t", header=T, as.is=T)[,c('id','set','num')]
tt = merge(tg, te, by='set', all.x=T)
tt$call[is.na(tt$call)] = ''
tt = tt[order(tt$id),c(2,1,3,4)]
df = tt
colnames(df) = c("id", "probe.set", "probe.number", "call.affy")
write.table(df, fo, row.names=F, col.names=T, sep="\t", quote=F)



dir = file.path(DIR_Misc2, "spada.affy", "Mtruncatula_3.5")
te1 = read.table(file.path(dir, "../mtdefl_pac.tbl"), sep="\t", header=T, as.is=T)
totalcall = apply(te1[,seq(6,206,by=8)], 1, function(x) "P" %in% x)
te = data.frame(set=te1$Probe.Set, call=totalcall)

fi = file.path(dir, "23.tbl")
#fi = file.path(dir, "33.tbl")
fo = file.path(dir, "25.tbl")
#fo = file.path(dir, "35.tbl")
tg = read.table(fi, sep="\t", header=T, as.is=T)[,c('id','set','num')]
tt = merge(tg, te, by='set', all.x=T)
tt$call[is.na(tt$call)] = ''
tt = tt[order(tt$id),c(2,1,3,4)]
df = tt
colnames(df) = c("id", "probe.set", "probe.number", "call.affy")
write.table(df, fo, row.names=F, col.names=T, sep="\t", quote=F)